pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -645,6 +650,45 @@ func parseDeviceAddress(addrString string) []string {
 	return addrs
 }
 
+func expandDiskImages(vmi *v1.VirtualMachineInstance, domain *api.Domain) {
+	logger := log.Log.Object(vmi)
+	for _, disk := range domain.Spec.Devices.Disks {
+		if imageCanBeExpanded(disk) {
+			logger.Infof("pre-start expansion of disk %v", disk)
+			err := expandDiskImage(disk.Source.File, disk.Size)
+			if err != nil {
+				logger.Reason(err).Error("failed to expand disk image")
+			}
+		}
+	}
+}
+
+func imageCanBeExpanded(disk api.Disk) bool {
+	diskInfo, err := converter.GetImageInfo(getSourceFile(disk))
+	if err != nil {
+		return false
+	}
+	if diskInfo.VirtualSize >= disk.Size {
+		return false
+	}
+	return true
+}
+
+func expandDiskImage(imagePath string, size int64) error {
```

Comment:
I think it would be a good idea to include "offline" somewhere in the name of functions that are intended to be used for offline resizing, and "online" in the other ones, to make it clear which one should be used in which context.

Reply:
done